shell/platform/windows/platform_handler_win32_unittests.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/platform_handler_win32.h"
+
+#include <memory>
+
+#include "flutter/shell/platform/common/json_method_codec.h"
+#include "flutter/shell/platform/windows/testing/mock_window_binding_handler.h"
+#include "flutter/shell/platform/windows/testing/test_binary_messenger.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "rapidjson/document.h"
+
+namespace flutter {
+namespace testing {
+
+namespace {
+using ::testing::_;
+
+static constexpr char kChannelName[] = "flutter/platform";
+
+static constexpr char kHasStringsClipboardMethod[] = "Clipboard.hasStrings";
+
+static constexpr char kTextPlainFormat[] = "text/plain";
+
+static constexpr char kValueKey[] = "value";
+static constexpr int kAccessDeniedErrorCode = 5;
+
+}  // namespace
+
+// A test version of the private ScopedClipboard.
+class TestScopedClipboard : public ScopedClipboardInterface {
+ public:
+  TestScopedClipboard();
+  ~TestScopedClipboard();
+
+  // Prevent copying.
+  TestScopedClipboard(TestScopedClipboard const&) = delete;
+  TestScopedClipboard& operator=(TestScopedClipboard const&) = delete;
+
+  int Open(HWND window);
+
+  bool HasString();
+
+  std::variant<std::wstring, int> GetString();
+
+  int SetString(const std::wstring string);
+
+ private:
+  bool opened_ = false;
+};
+
+TestScopedClipboard::TestScopedClipboard() {}
+
+TestScopedClipboard::~TestScopedClipboard() {
+  if (opened_) {
+    ::CloseClipboard();
+  }
+}
+
+// Always fail to open with kAccessDeniedErrorCode.
+int TestScopedClipboard::Open(HWND window) {
+  return kAccessDeniedErrorCode;
+}
+
+bool TestScopedClipboard::HasString() {
+  return true;
+}
+
+std::variant<std::wstring, int> TestScopedClipboard::GetString() {
+  return -1;
+}
+
+int TestScopedClipboard::SetString(const std::wstring string) {
+  return -1;
+}
+
+class MockMethodResult : public MethodResult<rapidjson::Document> {
+ public:
+  MOCK_METHOD1(SuccessInternal, void(const rapidjson::Document*));
+  MOCK_METHOD3(ErrorInternal,
+               void(const std::string&,
+                    const std::string&,
+                    const rapidjson::Document*));
+  MOCK_METHOD0(NotImplementedInternal, void());
+};
+
+TEST(PlatformHandlerWin32, HasStringsAccessDeniedReturnsFalseWithoutError) {

```

Comment:
We should add other test cases now that this code is being made testable. E.g., testing that other errors are still passed through, since that would have been easy to accidentally break in this PR. (Also that normal success cases work.)